module/mempool/consensus/execution_tree.go

```diff
@@ -84,6 +84,16 @@ func (et *ExecutionTree) getEquivalenceClass(result *flow.ExecutionResult, block
 	return vertex.(*ReceiptsOfSameResult), nil
 }
 
+func (et *ExecutionTree) HasReceipt(receipt *flow.ExecutionReceipt) bool {
+	et.RLock()
+	defer et.RUnlock()
+	vertex, found := et.forest.GetVertex((&receipt.ExecutionResult).ID())
```

Comment:
can we compute the ID before the read lock?

Reply:
+100